### PR TITLE
media-tv/kodi: adding cross-compilation support

### PR DESCRIPTION
**Pull request by g-braeunlich**

### CODE REVIEW

File: media-tv/kodi/kodi-16.0-r1.ebuild [hunk 1]
```diff
@@ -228,7 +227,11 @@ src_configure() {
 		$(use_enable midi mid) \
 		$(use_enable mysql) \
 		$(use_enable nfs) \
+		$(use_enable non-free) \
```
Comment (aballier) [inline media-tv/kodi/kodi-16.0-r1.ebuild]: what does this do, exactly ?

File: media-tv/kodi/kodi-16.0-r1.ebuild [hunk 2]
```diff
@@ -228,7 +227,11 @@ src_configure() {
...
 		$(use_enable opengl gl) \
+		$(use_enable openmax) \
+		$(use_enable optical-drive) \
+		$(use_enable optimizations) \
```
Comment (aballier) [inline media-tv/kodi/kodi-16.0-r1.ebuild]: this just plays with cflags so should stay disabled

File: media-tv/kodi/kodi-16.0-r1.ebuild [hunk 3]
```diff
@@ -228,7 +227,11 @@ src_configure() {
 		$(use_enable midi mid) \
 		$(use_enable mysql) \
 		$(use_enable nfs) \
+		$(use_enable non-free) \
 		$(use_enable opengl gl) \
+		$(use_enable openmax) \
+		$(use_enable optical-drive) \
```
Comment (aballier) [inline media-tv/kodi/kodi-16.0-r1.ebuild]: this seems to control 'cdio' support

File: media-tv/kodi/kodi-16.0-r1.ebuild [hunk 4]
```diff
@@ -228,7 +227,11 @@ src_configure() {
 		$(use_enable midi mid) \
 		$(use_enable mysql) \
 		$(use_enable nfs) \
+		$(use_enable non-free) \
 		$(use_enable opengl gl) \
+		$(use_enable openmax) \
```
Comment (aballier) [inline media-tv/kodi/kodi-16.0-r1.ebuild]: this probably has very few chances to work without adding the proper deps

**Comment by g-braeunlich** [comment]
Which changes are you referring to?
As you are mentioning multijob_init & multijob_finish, I guess you mean the changes in cross-compiler support. There we need in addition to the stuff in between multijob_init & multijob_finish (this is for the target system) JsonBuilder / TexturePacker for the build system, so that the tools can be used for the compilation process.
If I leave it out, cross-compilation fails.

**Comment by g-braeunlich** [comment]
I removed the commit for the new use flags as I don't have the time to investigate the deps of optical-drive, openmax and non-free. Maybe I will come back to it later (without optimizations use flag of course).

**Comment by g-braeunlich** [comment]
While testing the 9999 ebuild, emerge failed at
DVDOverlayCodecFFmpeg.cpp: In member function ‘virtual CDVDOverlay* CDVDOverlayCodecFFmpeg::GetOverlay()’:
DVDOverlayCodecFFmpeg.cpp:227:14: error: ‘AVSubtitleRect’ has no member named ‘data’
     if (rect.data[0] == NULL)
              ^
DVDOverlayCodecFFmpeg.cpp:279:23: error: ‘AVSubtitleRect’ has no member named ‘data’
     uint8_t* s = rect.data[0];
                       ^
DVDOverlayCodecFFmpeg.cpp:284:17: error: ‘AVSubtitleRect’ has no member named ‘linesize’
       s += rect.linesize[0];
                 ^
In file included from DVDOverlayCodecFFmpeg.cpp:27:0:
DVDOverlayCodecFFmpeg.cpp:289:63: error: ‘AVSubtitleRect’ has no member named ‘data’
       overlay->palette[i] = Endian_SwapLE32(((uint32_t *)rect.data[1])[i]);
                                                               ^
/var/tmp/portage/media-tv/kodi-9999/work/kodi-9999/xbmc/utils/EndianSwap.h:89:29: note: in definition of macro ‘Endian_SwapLE32’
 #define Endian_SwapLE32(X) (X)
                             ^
But I don't think, this is caused by my changes.

**Comment by aballier** [comment]
> DVDOverlayCodecFFmpeg.cpp:227:14: error: ‘AVSubtitleRect’ has no member named ‘data’

yes this is most likely it needs ffmpeg-3+

**Comment by aballier** [comment]
> Which changes are you referring to?
As you are mentioning multijob_init & multijob_finish, I guess you mean the changes in cross-compiler support. There we need in addition to the stuff in between multijob_init & multijob_finish (this is for the target system) JsonBuilder / TexturePacker for the build system, so that the tools can be used for the compilation process.
If I leave it out, cross-compilation fails.



it was supposed to be inlined in the diff, but i cannot see it now (i'd blame github)


but yes, that was those changes, but how does it fail exactly ?
from what i read from the build system, it should already build stuff in depends with host CC

**Comment by g-braeunlich** [comment]
In this case, g++ gets passed the wrong compiler flags in src_configure:

make: Entering directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/TexturePacker'
make -C ../libsquish-native
make[1]: Entering directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native'
rm -rf native/*; mkdir -p native
cd native; cp -a /usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/src/* .
cd native; sed "s|@PREFIX@|/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/squish-install|" config.in > config
cd native; mkdir -p /usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/squish-install/include; mkdir -p /usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/squish-install/lib/pkgconfig
make -C native
make[2]: Entering directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/native'
g++  -I. -O2 -DNDEBUG=1 -O2 -pipe -march=armv6j -mfpu=vfp -mfloat-abi=hard -I/usr/armv6j-hardfloat-linux-gnueabi/opt/vc/include -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -std=gnu++11 -oalpha.o -c alpha.cpp
g++: error: unrecognized command line option '-mfpu=vfp'
g++: error: unrecognized command line option '-mfloat-abi=hard'
Makefile:26: recipe for target 'alpha.o' failed
make[2]: *** [alpha.o] Error 1
make[2]: Leaving directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native/native'
Makefile:36: recipe for target 'native/libsquish.a' failed
make[1]: *** [native/libsquish.a] Error 2
make[1]: Leaving directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/libsquish-native'
Makefile:42: recipe for target '../libsquish-native/.installed-native' failed
make: *** [../libsquish-native/.installed-native] Error 2
make: Leaving directory '/usr/armv6j-hardfloat-linux-gnueabi/tmp/portage/media-tv/kodi-16.1/work/xbmc-16.1-Jarvis/tools/depends/native/TexturePacker'
configure: error: "Texturepacker not found"

**Comment by g-braeunlich** [comment]
Regarding ffmpeg, you were right!
Also, rtmp and joystick configure options have been removed and the themes now use other fonts.
The 9999 ebuild now works for me for ~amd64.

**Comment by SoapGentoo** [comment]
@g-braeunlich please rebase on latest masterm kodi changed a bit.

**Pull request closed without merging**